bootstraptest/test_yjit.rb

```diff
@@ -1,3 +1,29 @@
+# regression test for arity check with splat
+assert_equal '[:ae, :ae]', %q{
+  def req_one(a_, b_ = 1) = raise
+
+  def test(args)
+    req_one *args
+  rescue ArgumentError
+    :ae
+  end
+
+  [test(Array.new 5), test([])]
+} unless defined?(RubyVM::RJIT) && RubyVM::RJIT.enabled? # Not yet working on RJIT
+
+# regression test for arity check with splat and send
+assert_equal '[:ae, :ae]', %q{
+  def two_reqs(a, b_, _ = 1) = a.gsub(a, a)
```

Comment:
I didn't know you could do that. The underscore argument... This is an argument that's being ignored but has a default value? 😅

Reply:
Normally all the parameters need to have different names, but `_` is an exception. For example, this parses fine:
```
$ ruby -ce 'def foo(_, _, _) = 1'
Syntax OK
```
It's probably useful for API stability maintenance.

I thought you could make a create a method that have fewer locals than parameters this way, but luckily no. Each `_` gets a local even though they're inaccessible from Ruby land, so these are not that special at the VM level.